ribasim_qgis/core/topology.py

```diff
@@ -16,9 +22,11 @@ def explode_lines(edge: QgsVectorLayer) -> None:
         # Avoid infinite recursion and stackoverflow
         edge.blockSignals(True)
         provider = edge.dataProvider()
+        assert provider is not None
 
         with edit(edge):
-            provider.deleteFeatures([f.id() for f in edge.getFeatures()])
+            edge_iterator = cast(Iterable[QgsFeature], edge.getFeatures())
```

Comment:
`getFeatures()` actually returns a `QgsFeatureIterator`, but somehow mypy doesn't understand that it has an `__iter__()` and `__next()__` function

Reply:
I figured mypy doesn't understand, because the `__iter__()` function of qgis returns an `Optional[QgsFeatureIterator`

```py
    def __iter__(self) -> typing.Optional['QgsFeatureIterator']: ...
```

Reply:
Maybe “assert is not None” before would help then?

Reply:
Yes the alternative would be:

```py
edge_iterator = edge.getFeatures().__iter__()
assert edge_iterator is not None
```

But I'm not sure if that really makes a difference here.

Reply:
I leave the decision to you